### PR TITLE
GH-550 Add missing tests for Panda Utilities

### PR DESCRIPTION
**Pull request by wamukat**
This PR may increase test coverage a little for Panda Utilities.

* org.panda_lang.utilities.commons.iterable.ArrayIterable
* org.panda_lang.utilities.commons.iterable.ReversedIterable
* org.panda_lang.utilities.commons.iterable.SingletonIterator
* org.panda_lang.utilities.commons.iterable.ResourcesIterable
* org.panda_lang.utilities.commons.iterable.CharArrayDistributor

related to issue #550.

### CODE REVIEW

File: panda-utilities/src/test/java/org/panda_lang/utilities/commons/iterable/ArrayIterableTest.java [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+import org.junit.jupiter.api.Test;
+
+import java.util.Iterator;
+
+class ArrayIterableTest {
+
+    @Test
+    void testArrayIterable() {
+        ArrayIterable<String> arrayIterable =
+                new ArrayIterable<>(new String[]{"a", "b", "c"});
```
Comment (dzikoysk) [inline panda-utilities/src/test/java/org/panda_lang/utilities/commons/iterable/ArrayIterableTest.java]: We don't limit line length to 80 characters. The recommended line length is 120, the hard limit is 160.

File: panda-utilities/src/test/java/org/panda_lang/utilities/commons/iterable/ArrayIterableTest.java [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+
```
Comment (dzikoysk) [inline panda-utilities/src/test/java/org/panda_lang/utilities/commons/iterable/ArrayIterableTest.java]: Tests contain a lot of doubled empty lines, there is no need to keep it that way. 
Keep declarations together or group them in related context.

**Review by dzikoysk — CHANGES_REQUESTED** [review]

**Comment by dzikoysk** [comment]
Thanks for another PR! I see some variations of method names in test cases, we should stick to one style. At first, I was naming them by the associated method names, because utilities contains a lot of methods without related context and it was easier for me to just generate them, but... after a while, I really like the `should` convention and I had plans to apply it also in Panda repository. Would you be able to use that convention?

As an example of  `should` convention, take a look at this file: https://github.com/dzikoysk/reposilite/blob/master/reposilite-backend/src/test/groovy/org/panda_lang/reposilite/utils/FilesUtilsTest.groovy

**Comment by wamukat** [comment]
@dzikoysk Thanks for the review.
I will resolve the two reviews and apply the 'should' convention.

**Comment by dzikoysk** [comment]
Thanks, I'll add Groovy to simplify tests and it will simplify this process :)

File: panda-utilities/src/test/java/org/panda_lang/utilities/commons/iterable/ArrayIterableTest.java (hunk 1 above)
Comment (wamukat) [reply panda-utilities/src/test/java/org/panda_lang/utilities/commons/iterable/ArrayIterableTest.java]: Please check the commit bee18e9 to make sure it's fixed.

File: panda-utilities/src/test/java/org/panda_lang/utilities/commons/iterable/ArrayIterableTest.java (hunk 2 above)
Comment (wamukat) [reply panda-utilities/src/test/java/org/panda_lang/utilities/commons/iterable/ArrayIterableTest.java]: Please check the commit bee18e9 to make sure it's fixed.

**Comment by wamukat** [comment]
I applied the 'should' convention to test methods, but I'm not good at English so please check the content.

File: panda-utilities/src/test/java/org/panda_lang/utilities/commons/iterable/ArrayIterableTest.java [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+
+import java.util.Iterator;
+
+class ArrayIterableTest {
+
+    @Test
+    void next_should_moves_to_the_next_element_and_return_it() {
```
Comment (dzikoysk) [inline panda-utilities/src/test/java/org/panda_lang/utilities/commons/iterable/ArrayIterableTest.java]: Should `move`, not `moves` (https://data.grammarbook.com/blog/uncategorized/when-to-add-s-to-a-verb-2/ or https://www.gallaudet.edu/tutorial-and-instructional-programs/english-center/grammar-and-vocabulary/five-ways-to-use-s-at-the-end-of-a-noun-or-verb)

Methods below also contain this typo, just review them once again.

File: panda-utilities/src/test/java/org/panda_lang/utilities/commons/iterable/CharArrayDistributorTest.java [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+    void getPrevious_should_return_previous_element_of_current_element() {
+        CharArrayDistributor charArrayDistributor = new CharArrayDistributor(PATTERN);
+
+        Assertions.assertEquals('a', charArrayDistributor.getPrevious());
+        charArrayDistributor.setIndex(2);
+        Assertions.assertEquals('b', charArrayDistributor.getPrevious());
+    }
+
+    @Test
+    void getPrevious_should_returns_the_previous_element_by_the_specified_number() {
```
Comment (dzikoysk) [inline panda-utilities/src/test/java/org/panda_lang/utilities/commons/iterable/CharArrayDistributorTest.java]: The same as `moves` case above, should be `return`

File: panda-utilities/src/test/java/org/panda_lang/utilities/commons/iterable/ResourcesIterableTest.java [hunk 5]
```diff
@@ -27,17 +27,26 @@
...
 
+        Assertions.assertTrue(iterator.hasNext());
         Assertions.assertEquals("a", iterator.next());
         Assertions.assertEquals("b", iterator.next());
         Assertions.assertEquals("c", iterator.next());
+        Assertions.assertFalse(iterator.hasNext());
     }
 
+    @Test
+    void construct_with_nothing_should_throws_exception() {
```
Comment (dzikoysk) [inline panda-utilities/src/test/java/org/panda_lang/utilities/commons/iterable/ResourcesIterableTest.java]: I'd change this name to `constructor_should_throw_exception_if_empty`

**Review by dzikoysk — CHANGES_REQUESTED** [review]

**Comment by dzikoysk** [comment]
That's fine :) I've also linked related article to verbs with `-s` suffix to help you understand it. I'll merge it tomorrow.

**Comment by wamukat** [comment]
Thanks for the review. I fixed them.
And thanks for the related article on the verbs with "s" suffix!

**Review by dzikoysk — APPROVED** [review]

**Pull request merged**